File: app/src/main/java/com/lostincontext/commons/list/SpacesItemDecoration.java
package com.lostincontext.commons.list;


import android.graphics.Rect;
import android.support.v7.widget.RecyclerView;
import android.view.View;

public class SpacesItemDecoration extends RecyclerView.ItemDecoration {
    private int space;
    private final int span;

    public SpacesItemDecoration(int space,
                                int span) {
        this.space = space / 2;
        this.span = span;
    }

    @Override
    public void getItemOffsets(Rect outRect,
                               View view,
                               RecyclerView parent,
                               RecyclerView.State state) {


        int position = parent.getChildAdapterPosition(view);
        int column = position % span;

        if (position >= span) {
            outRect.top = space;
        }

        outRect.left = column * space / span;
        outRect.right = space - (column + 1) * space / span;


    }

    private static final String TAG = SpacesItemDecoration.class.getSimpleName();
}

File: app/src/main/java/com/lostincontext/commons/list/SpacesItemDecoration.kt
package com.lostincontext.commons.list


import android.graphics.Rect
import android.support.v7.widget.RecyclerView
import android.view.View

class SpacesItemDecoration(space: Int,
                           private val span: Int) : RecyclerView.ItemDecoration() {

    private val space: Int = space / 2


    override fun getItemOffsets(outRect: Rect,
                                view: View,
                                parent: RecyclerView,
                                state: RecyclerView.State?) {


        val position = parent.getChildAdapterPosition(view)
        val column = position % span

        if (position >= span) {
            outRect.top = space
        }

        outRect.left = column * space / span
        outRect.right = space - (column + 1) * space / span

    }

}